sw/device/lib/gpio.c

```diff
@@ -6,30 +6,28 @@
 
 #include "sw/device/lib/base/memory.h"
 #include "sw/device/lib/common.h"
+#include "sw/device/lib/dif/dif_gpio.h"
 #include "sw/device/lib/runtime/hart.h"
 
-void gpio_init(uint32_t oe) { REG32(GPIO_DIRECT_OE(0)) = oe; }
+#define GPIO0_BASE_ADDR 0x40010000
```

Comment:
I think we should drop `sw/device/lib/gpio.c` and switch callers over to the DIF, what do you think?

Reply:
Sounds good to me. I just wasn't sure if we wanted to keep this for consistency with other DIFs/drivers or not. (Dropping `gpio.c` may also require changes in other files.) @lenary, any comments?

Reply:
Removed the existing GPIO driver and updated callers to use the new GPIO DIF.